http/thumbnail.go

```diff
@@ -0,0 +1,92 @@
+package http
+
+import (
+	"github.com/disintegration/imaging"
+	"github.com/filebrowser/filebrowser/v2/errors"
+	"github.com/filebrowser/filebrowser/v2/files"
+	"image"
+	"image/gif"
+	"image/jpeg"
+	"image/png"
+	"mime"
+	"net/http"
+	"net/url"
+)
+
+var thumbnailHandler = withUser(func(w http.ResponseWriter, r *http.Request, d *data) (int, error) {
+	if !d.user.Perm.Download {
+		return http.StatusAccepted, nil
+	}
+
+	file, err := files.NewFileInfo(files.FileOptions{
+		Fs:      d.user.Fs,
+		Path:    r.URL.Path,
+		Modify:  d.user.Perm.Modify,
+		Expand:  true,
+		Checker: d,
+	})
+	if err != nil {
+		return errToStatus(err), err
+	}
+
+	if file.IsDir || file.Type != "image" {
+		return http.StatusNotFound, nil
+	}
+
+	return thumbnailFileHandler(w, r, file)
+})
+
+func thumbnailFileHandler(w http.ResponseWriter, r *http.Request, file *files.FileInfo) (int, error) {
+	fd, err := file.Fs.Open(file.Path)
+	if err != nil {
+		return errToStatus(err), err
+	}
+	defer fd.Close()
+
+	if r.URL.Query().Get("inline") == "true" {
+		w.Header().Set("Content-Disposition", "inline")
+	} else {
+		// As per RFC6266 section 4.3
+		w.Header().Set("Content-Disposition", "attachment; filename*=utf-8''"+url.PathEscape(file.Name))
+	}
+
+	srcImg, err := imaging.Decode(fd, imaging.AutoOrientation(true))
+	if err != nil {
+		return errToStatus(err), err
+	}
+	dstImg := fitResizeImage(srcImg)
+	w.Header().Add("Content-Type", mime.TypeByExtension(file.Extension))
+	err = func() error {
+		switch file.Extension {
+		case ".jpg", ".jpeg":
+			return jpeg.Encode(w, dstImg, nil)
+		case ".png":
+			return png.Encode(w, dstImg)
+		case ".gif":
+			return gif.Encode(w, dstImg, nil)
+		default:
+			return errors.ErrNotExist
+		}
+	}()
+	if err != nil {
+		return errToStatus(err), err
+	}
+	return 0, nil
+}
+
+const maxSize = 1080
```

Comment:
It doesn't make sense to resize an image to 1080px and display it as a thumbnail

Reply:
When I resize the image resolution, I can effectively reduce the size of the picture, for example reduce a 5MB picture to 500KB.

Reply:
If the actual size of the thumbnail is 100x100 for example then why we should return 1080x1080 image for that?

Reply:
In fact when the width and height < 1080,will return raw image,as below:
```go
func fitResizeImage(srcImage image.Image) image.Image {
	width := srcImage.Bounds().Dx()
	height := srcImage.Bounds().Dy()
	if width > maxSize && width > height {
		width = maxSize
		height = 0
	} else if height > maxSize && height > width {
		width = 0
		height = maxSize
	} else {
		return srcImage
	}
	return imaging.Resize(srcImage, width, height, imaging.NearestNeighbor)
}
```

Reply:
I still do not understand why we should transfer additional data over the network and resize the image in the browser? Since the thumbnail size is fixed on ui, let's do the same on the backend

Reply:
What do you think of the following changes?
1. add `/preview/thumb/file` API for list icon,fixed size 128*128,like this:
![image](https://user-images.githubusercontent.com/13160176/84357178-d2323900-abf7-11ea-8d05-78dccdabddfb.png)
2. change `/thumbnail/file` API to `/preview/big/file`  for review, other unchanged.

Reply:
I think we can use single handler with path param `/preview/{size}`.

Reply:
Of course,and the most important thing is to preview the picture max size, or keep the 1080px?

Reply:
Let's keep it 1080px. User can always download original one.

Reply:
Hi,I have a new commit,please review again.